public/components/FeatureAnywhereContextMenu/AssociatedDetectors/containers/AssociatedDetectors.tsx

```diff
@@ -0,0 +1,347 @@
+      ...confirmModalState,
+      isOpen: false,
+    });
+  };
+
+  const handleConfirmModal = () => {
+    setConfirmModalState({
+      ...confirmModalState,
+      isRequestingToClose: true,
+    });
+  };
+
+  const getDetectors = async () => {
+    dispatch(getDetectorList(GET_ALL_DETECTORS_QUERY_PARAMS));
+  };
+
+  // TODO: this part is incomplete because it is pending on a different PR that will have all the associate existing changes
+  const onAssociateExistingDetector = async () => {
+    console.log('inside create anomaly detector');
+  };
+
+  const handleUnlinkDetectorAction = (detector: DetectorListItem) => {
+    setDetectorToUnlink(detector);
+    if (!isEmpty(detector)) {
+      setConfirmModalState({
+        isOpen: true,
+        action: ASSOCIATED_DETECTOR_ACTION.UNLINK,
+        isListLoading: false,
+        isRequestingToClose: false,
+        affectedDetector: detector,
+      });
+    } else {
+      core.notifications.toasts.addWarning(
+        'Make sure selected detector has not been deleted'
+      );
+    }
+  };
+
+  const columns = useMemo(
+    () => getColumns({ handleUnlinkDetectorAction }),
+    [handleUnlinkDetectorAction]
+  );
+
+  const renderEmptyMessage = () => {
+    if (isLoading) {
+      return 'Loading detectors...';
+    } else if (!isEmpty(selectedDetectors)) {
+      return (
+        <EmptyAssociatedDetectorFlyoutMessage
+          isFilterApplied={true}
+          embeddableTitle={embeddableTitle}
+        />
+      );
+    } else {
+      return (
+        <EmptyAssociatedDetectorFlyoutMessage
+          isFilterApplied={false}
+          embeddableTitle={embeddableTitle}
+        />
+      );
+    }
+  };
+
+  const tableProps = {
+    items: selectedDetectors,
+    columns,
+    search: {
+      box: {
+        disabled: selectedDetectors.length === 0,
+        incremental: true,
+        schema: true,
+      },
+    },
+    hasActions: true,
+    pagination: true,
+    sorting: true,
+    message: renderEmptyMessage(),
+  };
+  return (
+    <div className="associated-detectors">
+      <EuiFlyout ownFocus size="m" paddingSize="m" onClose={closeFlyout}>
+        <EuiFlyoutHeader hasBorder>
+          <EuiTitle size="s">
+            <h2 id="associated-detectors__title">
+              Associated anomaly detectors
+            </h2>
+          </EuiTitle>
+        </EuiFlyoutHeader>
+        <EuiFlyoutBody style={{ overflowY: 'auto' }}>
+          {getUnlinkConfirmModal()}
+          <EuiFlexGroup alignItems="center">
+            <EuiFlexItem component="span">
+              <EuiTitle size="xxs">
+                <h3>{embeddableTitle}</h3>
+              </EuiTitle>
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiButton
+                fill
+                iconType="link"
+                onClick={() => {
+                  onAssociateExistingDetector();
+                }}
+              >
+                Associate a detector
+              </EuiButton>
```

Comment:
Should we add an `isLoading` prop to the button here so that users cant click on it if there is still data being loaded?

Reply:
Hmm I don't think in this case we need an isLoading since this is just to send them to the associate with existing detector flyout. It wont associate a detector so it has no direct dependency to the data being loaded. The only use case I can see is that if user clicks on this cause they see they don't have any detectors currently associated with the viz but that would have a loading component there. Users might want to still go to associate an existing detector flyout while all the current ones are loading. I can bring this up with UX team though.